### PR TITLE
boards: arm: stm32f3_seco_d23: Add SECO JUNO SBC-D23 board

### PR DESCRIPTION
**Pull request by ektor5**
Add support for SECO JUNO board embedded microcontroller STM32F302VC, designed to extend capabilities of the main processor Rockchip PX30. The microcontroller provides several interfaces, such as 2 and 4-Wire UARTs, USB, CAN, Modbus, 8-channels Timer, SPI, I2C and GPI/Os. The communication between the two processors is realized with an internal SPI line.

Based on stm32f3_disco board.

Signed-off-by: Ettore Chimenti <ettore.chimenti@seco.com>

### CODE REVIEW

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (gmarull) [inline boards/arm/stm32f3_seco_d23/doc/index.rst:27]: nit: avoid setting manual width/height, just provide a board picture that has the right size (current picture has black background, should be removed)

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (gmarull) [inline boards/arm/stm32f3_seco_d23/doc/index.rst:214]: `arm` seems wrong here

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (gmarull) [inline boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts:32]: nit: mixes tabs/spaces

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (gmarull) [inline boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts:76]: where is this alias used?

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (gmarull) [inline boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts:84]: does the board come with an embedded stlink?

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig
Comment (gmarull) [inline boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig:30]:
```suggestion
```

File: boards/arm/stm32f3_seco_d23/support/openocd.cfg
Comment (gmarull) [inline boards/arm/stm32f3_seco_d23/support/openocd.cfg:1]: should likely not rely on wrong board script

**Review by gmarull — CHANGES_REQUESTED** [review]
a few nitpicks, thanks for this contribution

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts:84]: No, clock is coming from an external oscillator. Removing.

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig:30]: Removing.

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:214]: Correcting

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:27]: Fixing and converting to png

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts:32]: fixing

File: boards/arm/stm32f3_seco_d23/support/openocd.cfg
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/support/openocd.cfg:1]: Replacing with actual configuration

**Comment by fabiobaltieri** [comment]
@ektor5 we don't use fixup patches in Zephyr PRs, patch series that represent the development history or aid reviewability are good, but review fixup should get amended to the original patches. In this case you probably want to squash the whole PR down into a single patch and force push it. Can you do that? The full development workflow documentation is: https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow

**Comment by ektor5** [comment]
@fabiobaltieri no problem, I thought it could be done from Github once approved.

**Comment by fabiobaltieri** [comment]
> @fabiobaltieri no problem, I thought it could be done from Github once approved.

It could in this case since this would be a single patch PR, but in general we want to preserve the (amended) development history.

**Comment by ektor5** [comment]
@fabiobaltieri Got it, thanks for the heads up. :+1:

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (gmarull) [inline boards/arm/stm32f3_seco_d23/doc/index.rst:58]:
```suggestion
More information about STM32F302VC can be found here:

- `STM32F302VC on www.st.com`_
- `STM32F302xC reference manual`_
```

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (gmarull) [inline boards/arm/stm32f3_seco_d23/doc/index.rst:96]:
```suggestion
```

redundant info

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (gmarull) [inline boards/arm/stm32f3_seco_d23/doc/index.rst:145]: mixes tabs/spaces. Is this really relevant? dts file is self-documented with pinctrl entries.

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig
Comment (gmarull) [inline boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig:30]: why is this needed?

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:145]: not everything is currently handled by the dts (e.g.: UART5 or the alternate config of some peripherals), but if it's useless I can remove it.

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig:30]: Not needed since `CONFIG_CAN_TRANSCEIVER_GPIO`: `depends on DT_HAS_CAN_TRANSCEIVER_GPIO_ENABLED`

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst]: done

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst]: tabs removed

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:157]: MDB stands for Modbus, if that's not clear I'll change it

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:162]: Already tested, I'll start a PR right away.

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts:161]: Yes, don't think it is necessary by default as not all configuration has it, but I don't think it can harm either.

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts:158]: I'll test and add it

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (fabiobaltieri) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:157]: Oh I see, yes please, the term is somewhat overloaded already (Metaware Debugger, Mandatory Data Byte...). But then what's the difference between UART2 for Modbus and UART3 with RS-485?

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (fabiobaltieri) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts:161]: Ok, I'd suggest leaving it enabled if possible, allows you to run usb samples with no other mods, gets ignored automatically if USB is not enabled.

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst]: They are attached to different transceivers: UART2 is connected to buffers and 5V translators, and UART3 is connected to a RS-422/RS-485 Interface that change signal into that protocol.

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts]: Done.

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:162]: https://github.com/zephyrproject-rtos/zephyr/pull/50385

Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:162]: @fabiobaltieri Should I wait for the MR to be merged and open a new one for adding the node in the board or can I add a commit in this MR?

Comment (fabiobaltieri) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:162]: Merged, you can drop this line.

**Review by fabiobaltieri — DISMISSED** [review]

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig
Comment (erwango) [inline boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig:10]: Please remove. Now extracted from device tree.

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig
Comment (erwango) [inline boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig:13]: Not sure this belongs to board configuration.

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts:111]: @gmarull got it, leaving them disabled
p.s: sorry for the late response

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts:155]: removing redundant entries.

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig:10]: Removing

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23_defconfig:13]: Removing

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:37]: I'm adding a tab for clearing that out

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:4]: Done

File: boards/arm/stm32f3_seco_d23/doc/index.rst
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/doc/index.rst:3]: Done

File: boards/arm/stm32f3_seco_d23/Kconfig.board
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/Kconfig.board:8]: It is 302. Indeed I sent an MR to solve this.

File: boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts
Comment (ektor5) [reply boards/arm/stm32f3_seco_d23/stm32f3_seco_d23.dts]: Added in #52342

**Comment by fabiobaltieri** [comment]
```
/__w/zephyr/zephyr/drivers/counter/counter_ll_stm32_rtc.c:453:2: error: #warning Kconfig COUNTER_RTC_STM32_CLOCK_LS* are deprecated. Please define clock source in dtsi file [-Werror=cpp]
  453 | #warning Kconfig COUNTER_RTC_STM32_CLOCK_LS* are deprecated. Please define clock source in dtsi file
      |  ^~~~~~~
```

From the CI, the STM32 RTC clock config has been updated since the last snapshot, see #50104 for the changes.

**Comment by ektor5** [comment]
RTC Clock fixed. Thanks @fabiobaltieri

**Review by FRASTM — DISMISSED** [review]

**Comment by erwango** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/52342 should be reviewed and merged first

**Comment by fabiobaltieri** [comment]
Heya, we don't use merge commits in Zephyr, I know the UI is confusing, unfortunately we can't disable the "update branch" button. Can you force push without the merge commit in the commit stack?

**Comment by ektor5** [comment]
@fabiobaltieri I'm really sorry, I wanted to do the rebase, I didn't realise it was going to merge instead. Will rebase manually and repush.

**Comment by fabiobaltieri** [comment]
> @fabiobaltieri I'm really sorry, I wanted to do the rebase, I didn't realise it was going to merge instead. Will rebase manually and repush.

Not a problem, the UI is confusing but we have no control over it, there's few feature requests already about this specific one but they seem to be stale, I'll look into adding a gitlint check about it.

**Comment by fabiobaltieri** [comment]
(https://github.com/zephyrproject-rtos/zephyr/pull/52342 merged, rebased it so the first commit get skipped from this pr)

**Review by fabiobaltieri — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Comment by gmarull** [comment]
@erwango ping

**Review by erwango — APPROVED** [review]

**Pull request merged**
